lib/Widgets/BackButton.vala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * BackButton is meant to be used in headers to navigate in
+ * {@link Adw.NavigationView}s. It will automatically detect when under a
+ * {@link Adw.NavigationView} and label itself with the title of the preceding page.
+ */
+[Version (since = "7.5.0")]
+public class Granite.BackButton : Gtk.Button {
+    /**
+     * A manually set label when used outside of {@link Adw.NavigationView}
+     */
+    public new string label { get; set; }
+
+    construct {
+        var image = new Gtk.Image.from_icon_name ("go-previous-symbolic");
+
+        var label_widget = new Gtk.Label ("");
+
+        var box = new Gtk.Box (HORIZONTAL, 0);
+        box.append (image);
+        box.append (label_widget);
+
+        child = box;
+
+        map.connect (on_map);
+        clicked.connect (on_click);
+
+        bind_property ("label", label_widget, "label");
+    }
+
+    private void on_map () {
+        var navigation_view = (Adw.NavigationView) get_ancestor (typeof (Adw.NavigationView));
+
+        if (navigation_view == null) {
+            return;
+        }
+
+        var navigation_page = (Adw.NavigationPage) get_ancestor (typeof (Adw.NavigationPage));
+
+        if (navigation_view == null) {
+            return;
+        }
+
+        var previous_page = navigation_view.get_previous_page (navigation_page);
+
+        if (previous_page != null) {
+            label = previous_page.title;
```

Comment:
Note that page label can and will change.

The logic in libadwaita's own back buttons is also a lot more complicated because navigation views can be nested, and can also be structured like:

navigation split view
  sidebar: navigation view
  content: navigation view

and that situation _is_ handled correctly

Reply:
Ah, tho the latter involves private api - forgot about that

Reply:
> Note that page title can and will change.

Fixed although I think that would be kinda questionable UX while the page itself is not visible?

> The logic in libadwaita's own back buttons is also a lot more complicated because navigation views can be nested, and can also be structured like:
> 
> navigation split view
> sidebar: navigation view
> content: navigation view
> 
> and that situation is handled correctly

I'm guessing that if the button is on the deepest view and there is no page left it still appears and instead navigates one above? 
I think for now and for our use case it might be enough to leave that out (since we also don't hide when there is no preceding page)

Reply:
> Should be fixed although I think that would be kinda questionable UX while the page itself is not visible?

Questionable yes, but I mean the property is writable, so...

> I'm guessing that if the button is on the deepest view and there is no page left it still appears and instead navigates one above?

Correct.

> I think for now and for our use case it might be enough to leave that out (since we also don't hide when there is no preceding page)

As you prefer

Reply:
Regarding the nested part cc @danirabbit ?